src/components/PopupWithDelete.js

```diff
@@ -0,0 +1,55 @@
+export default class PopupWithDelete {
+  constructor(modalSelector) {
+    this._modal = document.querySelector(modalSelector); // Modal element
+    this._submitButton = this._modal.querySelector("#card-delete-submit-btn"); // Submit button inside modal
+    this._setSubmitAction = () => {}; // Placeholder for the submit action function
+    this._closeButton = this._modal.querySelector(".profile-modal-close-btn"); // Assuming close button selector is correct
```

Comment:
`    this._closeButton = this._modal.querySelector(".profile-modal-close-btn"); // Assuming close button selector is correct
`

Please, delete it

`PopupWithDelete` should have `extends Popup` to get the parent methods and its listeners